### DISCUSSION TITLE
Battery problems on hue wall switch module

### DISCUSSION THREAD

**Discussion by fuNSA** [Koenkk/zigbee2mqtt#12545] [Devices]
Hey everybody,
does anyone else have problems with the battery life of the philips hue wall switch modules. I have 3 of them installed in which the battery only lasts about a month.

Thanks for your input!

Running:
zigbee2mqtt 2.14.0 (docker) on Raspberry Pi 3B+ 
SONOFF Zigbee 3.0 USB Dongle Plus (CC2652P)

**Comment by jrmax** [comment]
Same here.
But I think this was not from the beginning. I changed the coordinator to a better one. I did this by not resetting the whole network, I just changed the coordinator. Could this be a reason?

**Reply by fuNSA** [reply]
I don't think that's the problem. I had the problem with two different coordinators. The funny thing is that I don't have the problem with all switches. One has been running for at least half a year and is still at 80%.

**Comment by fuNSA** [comment]
I still have the problem. Changed the batteries 4 days ago and today they are empty again.

zigbee2mqtt 1.28.4 (docker) on Raspberry Pi 3B+
SONOFF Zigbee 3.0 USB Dongle Plus (CC2652P)

**Comment by theelog** [comment]
Running Zigbee2MQTT version 1.30.1-1 with Hue Wall Switches.
Batteries drain within a few weeks while they are rated for 5 years if connected to Hue Bridge.

**Comment by GijsvanDulmen** [comment]
Same issue over here on one of the modules. Funny fact. The module is like 1 meter away from the coordinator. So can't be connectivity issues (i guess).

**Comment by fannoute** [comment]
Hi,
Anyone found a solution ? I'm facing the issue but did not found any parameters that solve the issue.
Cheers

**Reply by jrmax** [reply]
Hi,
unfortunately not. I have replaced the switch that had this problem with one that is powered by the mains voltage.
I'm sorry, maybe someone else has found a solution.

**Reply by fuNSA** [reply]
Hi,
no unfortunately I haven't found a solution yet. It's very annoying to constantly have to change the batteries.

**Reply by fannoute** [reply]
Thanks for the reply. I will try to remove and reset the module to see if it improve something but it's very strange as all my aqara modules are with the same battery since years... 
zigbee2mqtt 1.30.4  (Home Assistant) on NUC 8i3BEK
SONOFF Zigbee 3.0 USB Dongle Plus (CC2652P)

**Comment by bjornarfjelldal** [comment]
Same issue here on one device, whereas another works fine. Hardware related..?

**Comment by martijn-brant** [comment]
Could there be a hardware short? Making it wake/press constantly? I have 10+ of these switches and haven;t seen this before.

**Comment by Branko75** [comment]
![Screenshot_2023-07-13-07-44-14-886_io homeassistant companion android](https://github.com/Koenkk/zigbee2mqtt/assets/125225601/c2d1207e-30b1-4cbc-8e5c-c6b53cb1703f)
![Screenshot_2023-07-13-07-43-51-812_io homeassistant companion android](https://github.com/Koenkk/zigbee2mqtt/assets/125225601/ab3f9384-21d6-4776-9bf4-f6d192cffc43)
last week I lost one yesterday another I have eight of them
zigbee2mqtt 1.32.1-1 (Home Assistant) on NUC
SONOFF Zigbee 3.0 USB Dongle Plus (CC2652P) latest firmware
before that happened I didn't even see them on the zigbee map but but they worked fine

**Comment by tomtom13** [comment]
Just FYI - I've got about 30 of old style ones (without hue button) and batteries last veeeerrryyyy long. Only issue I've had with the batteries was that I was purchasing batteries from amazon and those were duds, they were reporting as 100% and after a week would drop to 30% and within few weeks would be dead. I started buying from local supermarket and those last without an issue.

**Reply by bjornarfjelldal** [reply]
This is for the hue wall switch, not the hue switch :)

**Reply by tomtom13** [reply]
Sorry, my bad. But maybe next time provide some more information about specific device model ? ;)

**Reply by bjornarfjelldal** [reply]
Yeah, totally understand it can be hard to distiguish between "hue wall switch" and "hue dimmer switch" at a glance :)

**Comment by EvertDekker** [comment]
Try Changing the report interval: https://github.com/Koenkk/zigbee2mqtt/discussions/13279#discussioncomment-6237243
This prevent draining battery by sending the battery report often.

**Reply by Koenkk** [reply]
@bjornarfjelldal  could you try this?

**Reply by bjornarfjelldal** [reply]
Hi, @Koenkk ! Yes, I even removed all reporting without avail.. I will try to free up time to sniff on the network as per your suggestion in the issue, unless you have other suggestions

**Comment by popy2k14** [comment]
Exact same issue on one of 24x RWL021 hue dimmer switches. (old ones without the hue button)
At first i also thought the batterys are bad.
But it's the same on this particular switch with any battery.

Would be nice if there is an solution/workaround.

**Reply by egio12** [reply]
I have the exact same problem, did you find any workaround?

**Reply by popy2k14** [reply]
Yes, in my case using quality Duracell from local store and re-pairing near the coordinator did the trick. Please be aware that also from Big A some brands are fake products.

**Reply by egio12** [reply]
Thanks for the reply. I will try to follow your suggestions.

**Comment by marazmarci** [comment]
@Koenkk, is there a way to set the battery reporting interval to the usual ~24 hours automatically when the device is paired/interviewed?

**Comment by bikerglen** [comment]
My brand new Hue Wall Switch Module dropped from 100 battery to 80 battery to 60 battery overnight. I just set the min report interval to 64000 and the max rep interval to 65000 and will watch it overnight. Mine is running firmware version 1.0.5 with a build date of 20210504. 

Since the Hue module firmware version looks a bit dated (2021), I'm going unpair it from my Zigbee2MQTT network, pair it to my Hue hub, try to update the firmware, then bring it back into my Zigbee2MQTT network.

The rest of my version info:

Zigbee2MQTT version [1.35.1] commit: [3c96204]
Coordinator type     zStack3x0
Coordinator revision    20210708
Frontend version    0.6.151
Zigbee-herdsman-converters version    18.9.0
Zigbee-herdsman version    0.30.0

Edit: couldn't find a firmware update for the wall switch module. It's back on z2m network now. We'll see what happens.

**Comment by bikerglen** [comment]
I'm running the nRF Sniffer on an nRF dongle and seeing an 802.15.4 Data Request from the wall switch module every 500 ms:

```
IEEE 802.15.4 Command, Dst: 0x0000, Src: 0x55b6
    Frame Control Field: 0x8863, Frame Type: Command, Acknowledge Request, PAN ID Compression, Destination Addressing Mode: Short/16-bit, Frame Version: IEEE Std 802.15.4-2003, Source Addressing Mode: Short/16-bit
        .... .... .... .011 = Frame Type: Command (0x3)
        .... .... .... 0... = Security Enabled: False
        .... .... ...0 .... = Frame Pending: False
        .... .... ..1. .... = Acknowledge Request: True
        .... .... .1.. .... = PAN ID Compression: True
        .... .... 0... .... = Reserved: False
        .... ...0 .... .... = Sequence Number Suppression: False
        .... ..0. .... .... = Information Elements Present: False
        .... 10.. .... .... = Destination Addressing Mode: Short/16-bit (0x2)
        ..00 .... .... .... = Frame Version: IEEE Std 802.15.4-2003 (0)
        10.. .... .... .... = Source Addressing Mode: Short/16-bit (0x2)
    Sequence Number: 131
    Destination PAN: 0x2189
    Destination: 0x0000
    Source: 0x55b6
    Command Identifier: Data Request (0x04)
```

It's followed by an unaddressed ack with a matching sequence number:

```
IEEE 802.15.4 Ack, Sequence Number: 131
    Frame Control Field: 0x0002, Frame Type: Ack, Destination Addressing Mode: None, Frame Version: IEEE Std 802.15.4-2003, Source Addressing Mode: None
        .... .... .... .010 = Frame Type: Ack (0x2)
        .... .... .... 0... = Security Enabled: False
        .... .... ...0 .... = Frame Pending: False
        .... .... ..0. .... = Acknowledge Request: False
        .... .... .0.. .... = PAN ID Compression: False
        .... .... 0... .... = Reserved: False
        .... ...0 .... .... = Sequence Number Suppression: False
        .... ..0. .... .... = Information Elements Present: False
        .... 00.. .... .... = Destination Addressing Mode: None (0x0)
        ..00 .... .... .... = Frame Version: IEEE Std 802.15.4-2003 (0)
        00.. .... .... .... = Source Addressing Mode: None (0x0)
    Sequence Number: 131
```

Since this is at the 802.15.4 layer, is it my ZBDongle-P that is creating the acks? Any chance the wall switch module is looking for an ack with an address rather than unaddressed ack?

**Reply by bikerglen** [reply]
I upgraded Sonoff dongle to Coordinator revision 20230507 and still see the requests and acks every 500 ms.

**Reply by marazmarci** [reply]
Just an idea: could you please try to move the Hue Wall Switch further away, so it's +1 hop away from the coordinator (through at least one Zigbee router device)?

**Reply by bikerglen** [reply]
This entire Zigbee network exists solely on my workbench. Lol. But maybe I can put a router device on an extension cord in the front yard then walk away from that carrying the hue wall switch module until it connects to the router and log all that traffic over the weekend.

I'm also going to remove the wall switch module from the workbench network, configure wireshark to monitor my separate hue network, then add it to the hue network. Hopefully, I can decrypt the traffic, capture all the pairing and configuration details, and see if it does the 500 ms data request ping with the hue hub too. If that solves the ping and battery issue, I may use the hue hub's streaming interface to get the button press / hold / release messages from the wall switch module instead of Z2M.

And in parallel with that, I'm building my own outdoor-rated, waterproof, battery-powered button using nRF and SiLabs Zigbee modules. Everything works on both platforms including Z2M integration except the low-current battery part.

One way or another, I'm getting a battery-powered, outdoor-rated Zigbee button. Probably should have just torn open a Sonoff SNZV-01P, soldered wires to it, shoved it in a waterproof box with a new button, and been done with it.....

**Reply by marazmarci** [reply]
This plan sounds good, thanks! Good luck with decrypting traffic from the Hue network! 🙈

**Reply by marazmarci** [reply]
Perhaps you are making an outdoor Zigbee button for ringing the doorbell? If yes, I'm interested 😄
My biggest concern is if a coin cell battery can survive a winter 🤔
But not sure if other battery types could handle it better, I don't have experience in this.

**Reply by marazmarci** [reply]
Oh, and I forgot to mention that I own a Philips Hue Wall Switch Module (for 1.5 months now), and its battery didn't drain for me, it's on ~80% since the beginning. It's a few Zigbee router hops away from my C2652P coordinator, that's why I suggested the same for you to try.

**Reply by bikerglen** [reply]
It's going to be a button for just inside the gate to turn the backyard landscape lighting on and off. It should work as a doorbell too. I'm not sure about the battery life but I'm using a 22mm button in a 70x70x50 mm enclosure so I have room to upgrade the batteries. We had some -20C days recently. It would have been good test weather.

My Hue wall switch module seems to be hanging at 80% over the month I've owned mine even with the constant pings but I'd like to get them to stop or slow down if possible. Especially if they don't do that with the official Hue hub.

**Reply by bikerglen** [reply]
It babbles just as much connected to my Hue hub. Since the destination address is 0x0005, it's probably running through one of the lights or plugs or something rather than directly to the coordinator. Guess I'll let it be for now and watch the battery over the next few months. At least it's not buried behind a switch plate and switches.

```
No.	Time	Source	Destination	Protocol	Length	Info
2187	88.385087	0x2bc0	0x0005	IEEE 802.15.4	10	Data Request
2230	88.892417	0x2bc0	0x0005	IEEE 802.15.4	10	Data Request
2267	89.394770	0x2bc0	0x0005	IEEE 802.15.4	10	Data Request
2311	89.900268	0x2bc0	0x0005	IEEE 802.15.4	10	Data Request
2357	90.406341	0x2bc0	0x0005	IEEE 802.15.4	10	Data Request
2413	90.912290	0x2bc0	0x0005	IEEE 802.15.4	10	Data Request
2434	91.415713	0x2bc0	0x0005	IEEE 802.15.4	10	Data Request
2485	91.921269	0x2bc0	0x0005	IEEE 802.15.4	10	Data Request
2515	92.427122	0x2bc0	0x0005	IEEE 802.15.4	10	Data Request
2548	92.934099	0x2bc0	0x0005	IEEE 802.15.4	10	Data Request
```

**Reply by marazmarci** [reply]
Is it every 500ms non-stop? Or does it stop eventually?

**Reply by bikerglen** [reply]
I moved it back to the Sonoff coordinator. It has stopped doing it on a regular basis. Now it does a few of them each time I press the button:

```
No.	Time	Source	Destination	Protocol	Length	Info
4	29.820581	0x0bd9	0x0000	ZigBee HA	56	Unknown Command: 0x00, Seq: 5
6	29.936680	0x0bd9	0x0000	ZigBee HA	56	Unknown Command: 0x00, Seq: 6
8	30.827398	0x0bd9	0x0000	IEEE 802.15.4	10	Data Request
10	31.835728	0x0bd9	0x0000	IEEE 802.15.4	10	Data Request
12	32.847671	0x0bd9	0x0000	IEEE 802.15.4	10	Data Request
15	37.896224	0x0bd9	0x0000	IEEE 802.15.4	10	Data Request
24	135.379983	0x0bd9	0x0000	ZigBee HA	56	Unknown Command: 0x00, Seq: 7
26	135.517242	0x0bd9	0x0000	ZigBee HA	56	Unknown Command: 0x00, Seq: 8
28	136.387882	0x0bd9	0x0000	IEEE 802.15.4	10	Data Request
30	137.399133	0x0bd9	0x0000	IEEE 802.15.4	10	Data Request
33	138.408067	0x0bd9	0x0000	IEEE 802.15.4	10	Data Request
35	143.456442	0x0bd9	0x0000	IEEE 802.15.4	10	Data Request
```

The above shows:
It was quiet. 
Then a button press, button release, and four requests.
Then another button press, another button release, and four requests.
Then quiet again.

This makes sense to me: Upon joining the network, stay awake and send a lot of data requests in case the coordinator has any configuration or data messages that need to be sent to the wall switch module. After some number of minutes / hours, go to sleep and only send data requests after a button press.

But I could have sworn it had been sending the data requests every 500 ms for days the last time I connected it to the Sonoff coordinator. Maybe it's behaving now? Strange... I need to try it with the official Hue hub again sometime.
